typed-racket-lib/typed-racket/types/resolve.rkt

```diff
@@ -61,12 +61,12 @@
                                     (free-identifier=? n (poly-name (current-poly-struct))))
        (define poly-num (length (poly-vars (current-poly-struct))))
        (if (= poly-num (length rands))
-           (when (not (or (ormap Error? rands)
-                          (andmap equal? rands
-                                  (poly-vars (current-poly-struct)))))
+           (unless (or (ormap Error? rands) (andmap equal? rands (poly-vars (current-poly-struct))))
              (tc-error (~a "structure type constructor applied to non-regular arguments"
-                           "\n  type: " rator
-                           "\n  arguments...: " rands)))
+                           "\n  type: "
+                           rator
+                           "\n  arguments...: "
+                           rands)))
```

Comment:
@sorawee A nice `fmt` improvement would be to keep description-value pairs adjacent in `~a` and similar formatting expressions.

Reply:
What exactly are considered "description-value pairs"?

Reply:
Some examples:

- The `field v ... ...` argument of `raise-arguments-error`.
- Alternating pairs of strings and expressions in the arguments to `~a`, `~v`, `~s`, etc.

That is, cases that construct the `<field>: <detail>` parts of an error message as described in [Error Message Conventions](https://docs.racket-lang.org/reference/exns.html#%28part._err-msg-conventions%29). Not sure how easy it is to detect all of these though.

Reply:
I agree that it's a good idea for the `raise-arguments-error` family and the `hash` family. I don't yet buy that the heuristic described for `~a` will work well.